tiledb/sm/query/readers/sparse_unordered_with_dups_reader.cc

```diff
@@ -1461,17 +1432,89 @@ SparseUnorderedWithDupsReader<BitmapType>::compute_fixed_results_to_copy(
   // later on. If not, add a partial tile at the end.
   if (cell_offset == max_num_cells ||
       cell_offsets.size() == result_tiles.size()) {
-    buffers_full_ = cell_offset == max_num_cells;
+    buffers_full = cell_offset == max_num_cells;
     cell_offsets.emplace_back(cell_offset);
   } else {
-    buffers_full_ = true;
+    buffers_full = true;
     cell_offsets.emplace_back(cell_offset);
-    cell_offsets.emplace_back(max_num_cells);
+
+    // For overlapping ranges, a cell might be included multiple times and we
+    // can only process it if we can include all of the values as the progress
+    // we save in the read state doesn't allow to track partial progress for a
+    // cell.
+    uint64_t rt_idx = cell_offsets.size() - 1;
+    auto rt = (ResultTileWithBitmap<BitmapType>*)result_tiles[rt_idx];
+    uint64_t min_pos = rt_idx == 0 ? first_tile_min_pos : 0;
+    uint64_t cells_to_copy = max_num_cells - cell_offset;
+
+    // Get the position of the cell that gets us to the desired number of
+    // cells.
+    uint64_t pos = rt->pos_with_given_result_sum(min_pos, cells_to_copy);
+
+    // Count the actual number of results.
+    uint64_t actual_cells_to_copy =
+        rt->result_num_between_pos(min_pos, pos + 1);
+
+    // If the last cell has a count > 1, it is possible to overflow the number
+    // of cells to copy. Don't include the last cell if that is the case.
+    if (cell_offset + actual_cells_to_copy > max_num_cells) {
+      actual_cells_to_copy = rt->result_num_between_pos(min_pos, pos);
+    }
+
+    // It is possible that the first cell of the partial tile doesn't fit. In
+    // that case, we don't include an extra cell offset.
+    if (actual_cells_to_copy != 0) {
+      cell_offsets.emplace_back(cell_offset + actual_cells_to_copy);
+    }
   }
 
   // Resize the result tiles vector.
   result_tiles.resize(cell_offsets.size() - 1);
 
+  return {buffers_full, cell_offsets};
+}
+
+template <class BitmapType>
+std::vector<uint64_t>
+SparseUnorderedWithDupsReader<BitmapType>::compute_fixed_results_to_copy(
```

Comment:
Same naming comment as above. (also a little unclear why we have to pass the `result_tiles` all the way down here, these two functions _seem_ separable at first glance. not necessary to address now)